library/Icingadb/Common/IcingaRedis.php

```diff
@@ -6,13 +6,61 @@
 
 use Exception;
 use Icinga\Application\Config;
-use Redis;
+use Icinga\Forms\ConfigForm;
+use Predis\Client as Redis;
 
 trait IcingaRedis
 {
     /** @var Redis Connection to the Icinga Redis */
     private $redis;
 
+    private function checkRedis()
+    {
+        $sections = [];
+        $config = new Config();
+
+        foreach (ConfigForm::transformEmptyValuesToNull($this->getValues()) as $sectionAndPropertyName => $value) {
+            if ($value !== null) {
+                list($section, $property) = explode('_', $sectionAndPropertyName, 2);
+                $sections[$section][$property] = $value;
+            }
+        }
+
+        foreach ($sections as $section => $options) {
+            $config->setSection($section, $options);
+        }
+
+        try {
+            $redis1 = $this->getPrimaryRedis($config);
+        } catch (Exception $e) {
+            $this->warning(sprintf(
+                t('Failed to connect to primary Redis: %s'),
+                $e->getMessage()
+            ));
+            return false;
+        }
+
+        if ($this->getLastIcingaHeartbeat($redis1) === null) {
+            $this->warning(t('Primary connection established but failed to verify Icinga is connected as well.'));
+            return false;
+        }
+
+        try {
+            $redis2 = $this->getSecondaryRedis($config);
+        } catch (Exception $e) {
+            $this->warning(sprintf(t('Failed to connect to secondary Redis: %s'), $e->getMessage()));
+            return false;
+        }
+
+        if ($redis2 !== null && $this->getLastIcingaHeartbeat($redis2) === null) {
+            $this->warning(t('Secondary connection established but failed to verify Icinga is connected as well.'));
+            return false;
+        }
+
+        $this->info(t('The configuration has been successfully validated.'));
+        return true;
+    }
```

Comment:
This was not part of the trait, why is it one now? This doesn't even make any sense, since this method can only be utilized if the trait is being used by a form. This implicit dependency doesn't suit in a trait which's usecase is not only for forms.

Remove that and keep it where it was.